qtdbusextended/dbusextendedabstractinterface.h

```diff
@@ -90,6 +98,94 @@ private Q_SLOTS:
     QDBusError m_lastExtendedError;
     bool m_propertiesChangedConnected;
 };
+
+template<class External, class Internal>
+External DBusExtendedAbstractInterface::internalPropGetExternal(const char *propname, Internal *propertyPtr, External convert(Internal from))
+{
+    m_lastExtendedError = QDBusError();
+
+    if (m_useCache) {
+        return convert(*propertyPtr);
+    }
+
+    if (m_sync) {
+        return qvariant_cast<External>(property(propname));
+    } else {
+        if (!isValid()) {
+            QString errorMessage = QStringLiteral("This Extended DBus interface is not valid yet.");
+            m_lastExtendedError = QDBusMessage::createError(QDBusError::Failed, errorMessage);
+            qDebug() << Q_FUNC_INFO << errorMessage;
+            return External();
+        }
+
+        int propertyIndex = metaObject()->indexOfProperty(propname);
+
+        if (-1 == propertyIndex) {
```

Comment:
Interesting choice to have value before the variable in comparison. It's usually the other way around so I wonder why it's like this here.

Reply:
This is a good question; I wouldn't usually do it this way round, but in this case it's mirroring the code in [DBusExtendedAbstractInterface](https://github.com/sailfishos/amber-mpris/blob/92df031f93b5900adb89fa750afbcf33e15afa6f/qtdbusextended/dbusextendedabstractinterface.cpp#L168), which was the basis for these methods.

This usage can be traced back to the [qtdbusextended](https://github.com/sailfishos/qtdbusextended/blob/eaf7e19515e40ddcfcdfe6beff1a708541349046/src/dbusextendedabstractinterface.cpp#L171) project, at which point I guess it was just preference.

My personal opinion is that it's better to mirror the usage, to retain the similarity between the two, but I could go through and change all of them?

Reply:
Yeah. I think it's fine, although it doesn't follow any convention that we've had.